samples/enroll.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+require_once 'vendor/autoload.php';
+
+use Dotenv\Dotenv;
+use Virgil\CryptoImpl\VirgilCrypto;
+use Virgil\PureKit\Protocol\Protocol;
+use Virgil\PureKit\Protocol\ProtocolContext;
+use Virgil\PureKit\Core\PHE;
+
+try {
+    printf("Starting enroll...\n");
+
+    // MAIN CONFIGURATION
+
+    $userTableExample = 'user_table.json';
+    $mainTableExample = 'main_table.json';
+
+    $recoveryPrivateKeyFile = "recovery_private_key.pem";
+
+    $virgilCrypto = new VirgilCrypto();
+    $phe = new PHE();
+
+    // INITIALIZE PUREKIT
+
+    // Set here your PureKit credentials
+    $env = (new Dotenv(".", ".env"))->load();
+
+    $context = (new ProtocolContext)->create([
+        'appSecretKey' => $_ENV['APP_SECRET_KEY'],
+        'appToken' => $_ENV['APP_TOKEN'],
+        'servicePublicKey' => $_ENV['SERVICE_PUBLIC_KEY'],
+        'updateToken' => '' // needs to be empty
+    ]);
+
+    $protocol = new Protocol($context);
+
+    // LOAD DATABASE
+
+    $userTableString = file_get_contents($userTableExample);
+    $userTable = json_decode($userTableString);
+
+    $mainTableString = file_get_contents($mainTableExample);
+    $mainTable = json_decode($mainTableString);
+
+    // GENERATE AND STORE RECOVERY KEYS
+
+    printf("Generating Recovery Keys\n");
+
+    $keyPair = $virgilCrypto->generateKeys();
+    $privateKey = $keyPair->getPrivateKey();
+    $publicKey = $keyPair->getPublicKey();
+
+    // Store exported keys:
+    $privateKeyExported = $virgilCrypto->exportPrivateKey($privateKey);
+    $publicKeyExported = $virgilCrypto->exportPublicKey($publicKey);
+
+    // Convert to PEM only for this sample
+    $privateKeyPEM = VirgilKeyPair::privateKeyToPEM($privateKeyExported);
+    printf("Storing Recovery Private Key to the $recoveryPrivateKeyFile file\n");
+
+    $fp = fopen('recovery_private_key.pem', 'w');
+    fwrite($fp, $privateKeyPEM);
+    fclose($fp);
+
+    printf("Storing Recovery Public Key to the main_table.json\n");
+    $mainTable[0]->recovery_public_key = VirgilKeyPair::publicKeyToPEM($publicKeyExported);
+
+    // ENROLL AND ENCRYPT USER ACCOUNTS
+
+    foreach ($userTable as $user) {
+        printf("Enrolling user '%s'", $user->username);
+
+        // Ideally, you'll ask for users to create a new password, but
+        // for this guide, we'll use existing password in DB
+        $enroll = $protocol->enrollAccount($user->passwordHash);
+
+        // Save record to database
+        $user->record = base64_encode($enroll[0]);
+
+        // Use encryptionKey for protecting user data & save in database
+        $encryptionKey = $enroll[1];
+        $user->ssn = base64_encode($phe->encrypt($user->ssn, $encryptionKey));
+
+        // Import keys:
+        $privateKeyImported = $virgilCrypto->importPrivateKey($privateKeyExported);
```

Comment:
What's the point of private key here? Private key should be kept some on some other PC, not on actual running server